src/compiler/checker.ts

```diff
@@ -6662,46 +6677,95 @@ namespace ts {
             return s.valueDeclaration ? s.valueDeclaration.kind : SyntaxKind.PropertyDeclaration;
         }
 
-        function getDeclarationFlagsFromSymbol(s: Symbol) {
+        function getDeclarationFlagsFromSymbol(s: Symbol): NodeFlags {
             return s.valueDeclaration ? getCombinedNodeFlags(s.valueDeclaration) : s.flags & SymbolFlags.Prototype ? NodeFlags.Public | NodeFlags.Static : 0;

```

Comment:
We use the pattern commonly, I do not see the point of changing the pattern now. I have actually find it easier to read when the condition is not spread on multiple checks. i am fine splitting it on two lines, but not switching the conditional operator.